hdc/algo/ops/zonal.py

```diff
@@ -6,7 +6,7 @@
 
 
 @lazycompile(njit)
-def do_mean(pixels, z_pixels, num_zones, nodata, z_nodata, dtype="float64"):
+def do_mean(pixels, z_pixels, num_zones, nodata, z_nodata, out_dtype=np.float32):
```

Comment:
why this change is necessary? don't see any matching call-site changes

Reply:
ok, I see why. Usually this problem of passing extra args to dask graph is best done with `functools.partial(my_func, **fixed_named_args_that_might_alias)`

Reply:
There were a couple of problems with the previous implementation:

- `dtype` is already a keyword to `map_blocks`, so the naming is ambiguous when passed in through the accessor (which could be done with `functools.partial` I suppose.
- second problem was numba: when passing in a dtype with string notation it would fail compiling. Not sure if that would have been solved by using `functools.partial`, but passing the object works fine